databricks/koalas/indexes.py

```diff
@@ -565,6 +568,8 @@ def name(self) -> Union[Any, Tuple]:
 
     @name.setter
     def name(self, name: Union[Any, Tuple]) -> None:
+        if not is_hashable(name):
+            raise TypeError("Index.name must be a hashable type")
```

Comment:
Why did this come back?

Reply:
Otherwise the below case doesn't raise Exception unlike pandas.

```python
>>> kidx = ks.Index([1, 2, 3])
>>> kidx.name = ["renamed"]

>>> pidx = pd.Index([1, 2, 3])
>>> p.name = ["renamed"]
Traceback (most recent call last):
...
TypeError: Int64Index.name must be a hashable type
```

Reply:
Just added related tests!

Reply:
In my local with the diff from the latest commit:

```
$ git diff
diff --git a/databricks/koalas/indexes.py b/databricks/koalas/indexes.py
index 7b01d4c..673caa1 100644
--- a/databricks/koalas/indexes.py
+++ b/databricks/koalas/indexes.py
@@ -568,8 +568,6 @@ class Index(IndexOpsMixin):

     @name.setter
     def name(self, name: Union[Any, Tuple]) -> None:
-        if not is_hashable(name):
-            raise TypeError("Index.name must be a hashable type")
         self.names = [name]

     @property
```

```py
>>> kidx = ks.Index([1, 2, 3])
>>> kidx.name = ["renamed"]
Traceback (most recent call last):
...
TypeError: Index.name must be a hashable type
```

Reply:
Roll back to the previous fix!